server/item/bow.go

```diff
@@ -46,27 +46,29 @@ func (Bow) Release(releaser Releaser, tx *world.Tx, ctx *UseContext, duration ti
 		return
 	}
 
-	arrow, ok := ctx.FirstFunc(func(stack Stack) bool {
-		_, ok := stack.Item().(Arrow)
+	var arrow Stack
+	if !left.Empty() {
+		if _, ok := left.Item().(Arrow); ok {
+			arrow = left
+		}
+	}
+
+	var ok bool
+	arrow, ok = ctx.FirstFunc(func(stack Stack) bool {
```

Comment:
This overwrites an arrow from the left hand

Reply:
Oh shoot I’m not checking if arrow is empty will fix